quickwit/quickwit-search/src/root.rs

```diff
@@ -730,26 +750,32 @@ pub(crate) async fn search_partial_hits_phase(
     split_metadatas: &[SplitMetadata],
     cluster_client: &ClusterClient,
 ) -> crate::Result<LeafSearchResponse> {
-    let leaf_search_responses: Vec<LeafSearchResponse> =
-        if is_metadata_count_request(search_request) {
-            get_count_from_metadata(split_metadatas)
+    let mut leaf_search_responses: Vec<LeafSearchResponse> =
+        Vec::with_capacity(split_metadatas.len());
+    let mut leaf_search_jobs = Vec::new();
+    for split in split_metadatas {
+        let start_time = split.time_range.as_ref().map(|x| x.start()).copied();
+        let end_time = split.time_range.as_ref().map(|x| x.end()).copied();
+        if is_metadata_count_request(search_request, start_time, end_time) {
```

Comment:
Nit, to make `search_partial_hits_phase` a bit more readable, I would pass the `time_range` directly to `is_metadata_count_request`